src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -40,6 +48,7 @@
 public class MLCommonsClientAccessor {
     private static final List<String> TARGET_RESPONSE_FILTERS = List.of("sentence_embedding");
     private final MachineLearningNodeClient mlClient;
+    private final Map<String, Boolean> modelAsymmetryCache = new ConcurrentHashMap<>();
```

Comment:
please add comment around this cache behavior and usage. You can start from following:
- it's local for the data node
- how we invalidate and evict if we ever going to do this
- if cache miss, what is the latency to do retrieve value via API call and put it to cache
- how big is a single object, that also implies some eviction strategy as we cannot grow indefinitely 
- what is the behavior is case of node restart/drop/model got redeployed

Reply:
* Yes, it is local for the data node.
* We are never evicting entries.
* The latency is that of fetching the model configuration and parsing it.
* The cache will map `modelId` to `Boolean`; the size requirement is 20 chars + 1 Boolean per entry.
* If the node restarts the cache will be empty; the first inference request will lead to the cache entry being populated
* if the node drops the cluster, the inference request will fail. The cache will not be emptied.
* if the model gets redeployed the user will need to request inference with a new `modelId`; which will lead to a new cache entry. The old one will continue there.

The only scenario I can think of where this design could be problematic is if a malicious actor floods the cluster with billions of inference requests with inexistent models. This would lead to an increase in heap usage that would never be GC'd. 

Let me know if you think this should be changed.

Reply:
@br3no It's fine since this not accessible from public API, to use huge amount of inexistent models, the malicious actor has to first got the permission to update the pipeline, and if that's true the malicious actor has much more straightforward means to attack the system.

Reply:
@br3no thank you for response, it all makes sense. 
I think we need to:
- use existing OpenSearch component for cache 
- set a limit on either size or time in the cache, that should lower the chance of data node going to the critical level of heap memory usage.

please take a look on these classes: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/cache/Cache.java and https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/cache/CacheBuilder.java

it has many nice things like multiple modes for eviction and stats

Reply:
I'm open to make the change but I don't think it's a mandatory, the cache is to store the model info and it only consumes very little space comparing to a model, so the heap memory could reach throttle when registering a new model instead of writing the little metadata into this cache. The only case is there's a way to bypass the model registration and put huge amount of info to the cache directly which is not a case as above explanation.

Reply:
Agreeing with @zane-neo , `modelAsymmetryCache` is just a map from model_id to boolean. The size is very limited. @br3no I think you just need to add some comments to make the code more readable.